datafusion/sqllogictest/test_files/decimal.slt

```diff
@@ -858,10 +858,11 @@ select log(2, 100000000000000000000000000000000000::decimal(38,0));
 116
 
 # log(10^35) for decimal128 with another base
+# TODO: this should be 116.267483321058, error with native decimal log impl
 query R
 select log(2.0, 100000000000000000000000000000000000::decimal(38,0));
 ----
-116.267483321058
+116
```

Comment:
This is interesting; we weren't actually making use of the native decimal implementation for log here apparently. The decimal value was actually being casted to float. Fixing the signature to have decimals take precedence (so they don't get casted to float) reveals a bug in our decimal log implementation.

Note to self: raise separate issue for this